Update/ignore user

nostrbuddha opened the pull request:
Features covered:
 - Fix https://github.com/bisq-network/bisq-mobile/issues/418
 - No messages, offers will be visible from 'Ignored users', except for current trade.
 - 'Ignore user' feature available in trade chat -> Chat's context menu
 - All 'Ignored users' are now listed in a new 'Ignored users' screen in settings. This screen has option to 'Undo ignore'

Issues:
 - Right now ignoring / undo ignoring takes into effect after App restart. Not immediate.
   - Same is conveyed in the ignore confirmation dialog.
   - When processing new offers, there is a `OfferServiceFacade :: isValidOfferMessage()`, which has logic for ignore user.
   - This logic could be made to be in effect immediately, if we lift the `isValidOfferMessage()` from `OfferServiceFacade` to `OfferbookPresenter`. Like to discuss before doing this.
   
Update:
 - Made ignore/undo realtime. Doesn't require app restart.
 - Added i18n/GeneratedResourceBundles* to .gitignore.
 - Settings -> Ignores profiles will be shown, only where is the ignored list.

Need this PR: https://github.com/bisq-network/bisq2/pull/3710 for xClients

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Ignored Users screen to view/manage ignored profiles and unblock with confirmation.

* **Improvements**
  * Chat now hides messages from ignored users; Settings menu updates to show Ignored Profiles.
  * Offerbook filters out offers from ignored users; Open Trade shows a warning when a peer is ignored.
  * Profile APIs expanded to support fetching profiles and managing ignore state.

* **UI/UX**
  * Confirmation dialogs for ignore/unblock flows; "report user" menu item removed.

* **Localization**
  * New i18n strings for ignore/unignore flows in multiple languages.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rodvar reviewed (changes requested):
First pass tested on Android Node only whilst the http-api PR gets reviewed

Good implementation, need a couple of things done to get it merge ready:

 1. Ignored Users remove feature needs also to have the warning dialog letting the user know that user will be still ignored until he restarts the app
 2. When ignoring a user in the middle of a trade, the ongoing trade could still be progressed on his side and progress will arrive to the peer. But progress done from the peer will be ignored. This brings all sort of trading potential issues, we need to either: Cancel/Reject the trade when the user decides to ignore a user or at least add a warning dialog in the ongoing trade letting the user know he is trading with a blocked user and preventing him from progressing the trade (only option is to cancel / reject)
  3. Nit: change "Ignore Users" setting to "Ignore Profiles" and sort right after "User Profile"
  
  let me know your thoughts / questions if any!

nostrbuddha commented on shared/domain/src/commonMain/kotlin/network/bisq/mobile/client/service/user_profile/ClientUserProfileServiceFacade.kt:
```diff
@@ -148,7 +151,20 @@ class ClientUserProfileServiceFacade(
     override suspend fun findUserIdentities(ids: List<String>): List<UserIdentityVO> {
-        val apiResult = apiGateway.findUserIdentities(ids)
+        return emptyList()
+    }
```
Removed this function, as its unused now

rodvar commented:
Regarding this in the description

```
Right now ignoring / undo ignoring takes into effect after App restart. Not immediate.
Same is conveyed in the ignore confirmation dialog.
When processing new offers, there is a OfferServiceFacade :: isValidOfferMessage(), which has logic for ignore user.
This logic could be made to be in effect immediately, if we lift the isValidOfferMessage() from OfferServiceFacade to OfferbookPresenter. Like to discuss before doing this.
```

what do you mean by lifting the function to a presenter level? can we just call the facade method in the presenter that needs it to make it work immediate?

rodvar reviewed (changes requested):
on Node it works much better now. Only comment would be in the settings, can we show the "Ignored Profiles" only when there is ignored profiles to show?

for the xclients I can't ignore the profile, clicking on the dialog "Ignore Profile" doesn't do anything and couldn't see any exceptions logged (testing with the latest api branch version) - is that still on the works? thanks buddha 🙏

nostrbuddha commented:
> what do you mean by lifting the function to a presenter level? can we just call the facade method in the presenter that needs it to make it work immediate?

Right now, the logic to ignore chat messages from Offer messages also happens inside `NodeOfferServiceFacade :: isValidOfferMessage()`. So it's better to leave that logic here and current flow as it is. It's different for `ClientOfferServiceFacade`.

But I can just move the `Ignored list check` logic to presenter level so it can happen on `OnViewAttached :: combine()`. This should make it real time.

Let me do this now.

nostrbuddha commented:
> for the xclients I can't ignore the profile, clicking on the dialog "Ignore Profile" doesn't do anything and couldn't see any exceptions logged (testing with the latest api branch version) - is that still on the works? thanks buddha 🙏

I tested this to be working. Did you also updated to the latest commit for the APIs @ https://github.com/bisq-network/bisq2/pull/3710. Because I changed some endpoints.

nostrbuddha commented:
> > for the xclients I can't ignore the profile, clicking on the dialog "Ignore Profile" doesn't do anything and couldn't see any exceptions logged (testing with the latest api branch version) - is that still on the works? thanks buddha 🙏
> 
> I tested this to be working. Did you also updated to the latest commit for the APIs @ [bisq-network/bisq2#3710](https://github.com/bisq-network/bisq2/pull/3710). Because I changed some endpoints.

Yes. There was a small issue. Fixed it and pushing now

nostrbuddha commented on shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/service/user_profile/UserProfileServiceFacade.kt:
```diff
@@ -64,14 +64,27 @@
+
+    suspend fun isUserIgnored(profileId: String): Boolean
+
+    suspend fun getIgnoredUserProfileIds(): List<String>
```
ignoredUserProfileIds: StateFlow<Set<String>> is not needed, since it’s only used in a simple screen with minimal user interaction.

rodvar commented on .gitignore:
```diff
@@ -31,3 +31,6 @@ credentials
 **/xcshareddata/WorkspaceSettings.xcsettings
+
+# i18n generated Resources
+**/i18n/GeneratedResourceBundles_*.kt
```
we can add this here but won't have any effect until we finish with #150 and remove the generated files from the repo as planned 👍

rodvar commented on androidNode/src/androidMain/kotlin/network/bisq/mobile/android/node/service/user_profile/NodeUserProfileServiceFacade.kt:
```diff
@@ -195,4 +195,32 @@ class NodeUserProfileServiceFacade(private val applicationService: AndroidApplic
+        }
+    }
+
+    override suspend fun undoIgnoreUserProfile(profileId: String) {
```
is it ok that this fails silently when failing?
how would the UI know to inform the user?

rodvar commented on shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/components/molecules/chat/ChatMessageContextMenu.kt:
```diff
@@ -80,14 +79,14 @@ fun ChatMessageContextMenu(
-                        onSetShowMenu(false)
-                    }
-                )
+//                DropdownMenuItem(
```
please add comment why we are leaving this commented dead code (feature to-be-implemented soon)

rodvar commented on shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/open_trades/selected/trade_chat/TradeChatPresenter.kt:
```diff
@@ -48,6 +48,9 @@ class TradeChatPresenter(
     val avatarMap: StateFlow<Map<String, PlatformImage?>> = _avatarMap
 
+    private val _ignoreUserId: MutableStateFlow<String> = MutableStateFlow("")
+    val ignoreUserId: StateFlow<String> = _ignoreUserId
```
a better way of linking StateFlow and MutableState flow is being discussed in #512 and there are PRs on the works to implement it across the board. The change is to avoid potential cast loopholes breaking immutability.

Would be nice to switch in this PR as well

rodvar commented on shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/settings/PaymentAccountSettingsScreen.kt:
```diff
@@ -53,13 +53,6 @@ fun PaymentAccountSettingsScreen() {
-    val menuTree: MenuItem = settingsPresenter.menuTree()
-    val menuPath = remember { mutableStateListOf(menuTree) }
-
-    RememberPresenterLifecycle(presenter, {
```
In general we need to have one call to RememberPresenterLifecycle on every view, is there a specific reason to avoid it here?

nostrbuddha replied:
Removed it by mistake. Fixed it now.

rodvar replied:
good we catch this one, it can bring difficult bugs

we might figure out a better way of integrating that call to every screen in the future (like composed composables and then make every screen use that base composable or sth like that

rodvar reviewed (changes requested):
Functionally is all working now in all 3 apps 😄 

Just a few minor tweaks/reviews we need to do and we are all done here.

 1. Please see screenshot below, ignored users card needs some love in terms of spacing/padding
 2. please check my code-related comments
 
 
<img width="338" height="147" alt="Screenshot 2025-08-11 at 09 14 19" src="https://github.com/user-attachments/assets/fc4ecac8-82b4-42cf-b188-4fc121645fd4" />

  
  thanks!

nostrbuddha commented on shared/domain/src/commonMain/kotlin/network/bisq/mobile/i18n/GeneratedResourceBundles_es.kt:
```diff
@@ -1542,6 +1542,7 @@ object GeneratedResourceBundles_es {
             "mobile.user.paymentAccounts.createAccount.paymentAccount.validations.minLength" to "ES_Min length: 3 characters",
             "mobile.components.button.passEitherTextOrCustomTextorIcon" to "ES_Error: Pass either text or customText or icon",
             "mobile.takeOffer.unexpectedError" to "ES_Unexpected error occurred, please try again",
+            "mobile.settings.ignoredUsers.empty" to "ES_ No ignored profiles",
```
This is all temporary. Will be fixed later

rodvar reviewed (approved):
ACK

💪
